vllm/v1/worker/gpu_model_runner.py

```diff
@@ -117,6 +118,32 @@ def __init__(
             dtype=self.dtype,
             device=self.device)
 
+        self.input_ids_cpu = torch.zeros(self.max_num_tokens,
+                                         dtype=torch.int32,
+                                         device="cpu",
+                                         pin_memory=self.pin_memory)
+        self.input_ids_np = self.input_ids_cpu.numpy()
+        self.positions_cpu = torch.zeros(self.max_num_tokens,
+                                         dtype=torch.int64,
+                                         device="cpu",
+                                         pin_memory=self.pin_memory)
+        self.positions_np = self.positions_cpu.numpy()
+        self.slot_mapping_cpu = torch.zeros(self.max_num_tokens,
+                                            dtype=torch.int32,
+                                            device="cpu",
+                                            pin_memory=self.pin_memory)
+        self.slot_mapping_np = self.slot_mapping_cpu.numpy()
```

Comment:
Does the resulting numpy here shares the memory buffer of the source tensor?

Reply:
> The returned ndarray and the tensor will share their storage, so changes to the tensor will be reflected in the ndarray and vice versa.

Yes. That's the trick here :)